### ISSUE TITLE
Incorrect line number for comments

### ISSUE THREAD

**Issue by idleman** [cherow/cherow#107]
When I run the following code:
```
import cherow from 'cherow';

const code =  `
  //  hello
  export const val = 123;
`;

const AST = cherow.parseModule(code, { comments: true, loc: true });
process.stdout.write(JSON.stringify(AST, null, '  '));

```
is the generated AST/loc information a bit strange ( see comments in the AST below):

```
{
  "type": "Program",
  "sourceType": "module",
  "body": [
    {
      "type": "ExportNamedDeclaration",
      "source": null,
      "specifiers": [],
      "declaration": {
        "type": "VariableDeclaration",
        "kind": "const",
        "declarations": [
          {
            "type": "VariableDeclarator",
            "init": {
              "type": "Literal",
              "value": 123,
              "loc": {
                "start": {
                  "line": 3,
                  "column": 31
                },
                "end": {
                  "line": 3,
                  "column": 34
                }
              }
            },
            "id": {
              "type": "Identifier",
              "name": "val",
              "loc": {
                "start": {
                  "line": 3,
                  "column": 25
                },
                "end": {
                  "line": 3,
                  "column": 28
                }
              }
            },
            "loc": {
              "start": {
                "line": 3,
                "column": 25
              },
              "end": {
                "line": 3,
                "column": 34
              }
            }
          }
        ],
        "loc": {
          "start": {
            "line": 3,
            "column": 19
          },
          "end": {
            "line": 3,
            "column": 35
          }
        }
      },
      "loc": {
        "start": {
          "line": 1,                               // <--- Should be 3 not 1
          "column": 0
        },
        "end": {
          "line": 3,
          "column": 35
        }
      }
    }
  ],
  "loc": {
    "start": {
      "line": 1,                                    // <--- Should be 3 not 1
      "column": 0
    },
    "end": {
      "line": 4,                                   // <--- Should be 3 not 4
      "column": 8
    }
  },
  "comments": [
    {
      "type": "SingleLine", 
      "value": "  hello\n",
      "start": 0,
      "end": 23,
      "loc": {
        "start": {
          "line": 1,                             // <--- Should be 2 not 1
          "column": 0
        },
        "end": {
          "line": 1,                            // <--- Should be 2 not 1
          "column": 0
        }
      }
    }
  ]
}
```
Is the result above expected?

**Node version**: v10.0.0 (with --experimental-modules flag)
**Cherow version**: 1.6.8

**Comment by fkleuver** [comment]
The program as a whole is 4 lines, and lines are 1 index-based. So start line 1 and end line 4 for the whole program makes sense to me. As for the rest, see comments below:

```
  "loc": {
    "start": {
      "line": 1,
      "column": 0
    },
    "end": {
      "line": 4,
      "column": 8 // <--- Perhaps this should be 0
    }
  },
```

```
  "comments": [
    {
      "type": "SingleLine", 
      "value": "  hello\n", // <--- Perhaps the \n shouldn't be part of the single-line comment here
      "start": 0,
      "end": 23,
      "loc": {
        "start": {
          "line": 1,                             // <--- Should be 2 not 1 (I agree)
          "column": 0
        },
        "end": {
          "line": 1,                            // <--- Should be 2 not 1 (I agree)
          "column": 0                       // <--- Perhaps this should be 23
        }
      }
    }
  ]
```

```
const code =  `/* line 1 */
/* line 2 */  //  hello
/* line 3 */  export const val = 123;
/* line 4 */`;
```

**Comment by KFlash** [comment]
@fkleuver agrew with you the comments are sliced from  current index position. And as long as itvexist a newline after the coment, this will be the result

**Comment by KFlash** [comment]
Will be fixed in v. 2.0

**Comment by KFlash** [comment]
This have been corrected in 2.0. So closing this for now.
